src/reduxStore/slices/batchTransactionsSlice.ts

```diff
@@ -0,0 +1,51 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { BatchTransactionsResponseType } from 'types';
+import { logoutAction } from '../commonActions';
+
+export interface BatchTransactionsSliceStateType {
+  [batchId: string]: BatchTransactionsResponseType;
+}
+
+const initialState: BatchTransactionsSliceStateType = {};
+
+export const batchTransactionsSlice = createSlice({
+  name: 'batchTransactionsSlice',
+  initialState,
+  reducers: {
+    setBatchTransactions: (
+      state: BatchTransactionsSliceStateType,
+      action: PayloadAction<BatchTransactionsResponseType>
+    ) => {
+      state[action.payload.id] = action.payload;
+    },
+    updateBatchTransactions: (
+      state: BatchTransactionsSliceStateType,
+      action: PayloadAction<BatchTransactionsResponseType>
+    ) => {
+      state[action.payload.id] = {
```

Comment:
state[action.payload.id] = action.payload;
why not use setBatchTransactions?

Reply:
There are 2 functions (set, update) just for conventional purposes. When a batch is created then `set` fn is called and when the batch is monitored and the store should be updated, then we will use `update` fn. For future upgrades, there is a big benefit to using this "separation of concern" approach